### DISCUSSION TITLE
Berserker skill

### DISCUSSION THREAD

**Discussion by JesterWizard** [MokhaLeee/fe8u-cskillsys-kernel#304] [General]
I want to add a skill called ``Berserker`` which will cause a player unit to operate under AI control but with ``Resolve`` stat boosts. 

I believe I need to make the edit here, but I can't see a way to do it without editing the function signature, as I only see the ``AIAttemptOffensiveAction`` function called inside a lyn file. Is there a better way to do it?

![image](https://github.com/user-attachments/assets/859fa74d-8873-46d8-b15e-22c2b9c7b855)

**Comment by MokhaLeee** [comment]
I don't understand the skill effect you want to achieve. You need to describe it in more detail.

**Reply by JesterWizard** [reply]
So basically, you know how the berserk staff works on player units? It makes them move under AI control. I want a similar effect for this skill. But it'll only work on player units, and they'll only target enemy units. 

Think Berserker from FF5 

https://finalfantasy.fandom.com/wiki/Berserker_(Final_Fantasy_V_job)

**Reply by MokhaLeee** [reply]
I see, but I think `AiAttemptOffensiveAction` should not be the target function, it is `isEnemy` that deside on judgement. Also you need to consider how to generate the berserk phase ai list.

**Reply by JesterWizard** [reply]
I already have a working version of this on my fork now so I'll close this.
